code/messaging/src/androidTest/java/com/adobe/marketing/mobile/MessagingFunctionalTestUtils.java

```diff
@@ -0,0 +1,277 @@
+        }
+        cacheManager.deleteFilesNotInList(new ArrayList<String>(), IMAGES_CACHE_SUBDIRECTORY);
+        cacheManager.deleteFilesNotInList(new ArrayList<String>(), MESSAGES_CACHE_SUBDIRECTORY);
+    }
+
+    /**
+     * Set the persistence data for Edge Identity extension.
+     */
+    static void setEdgeIdentityPersistence(final Map<String, Object> persistedData) {
+        if (persistedData != null) {
+            final JSONObject persistedJSON = new JSONObject(persistedData);
+            updatePersistence("com.adobe.edge.identity",
+                    "identity.properties", persistedJSON.toString());
+        }
+    }
+
+    /**
+     * Helper method to update the {@link SharedPreferences} data.
+     *
+     * @param datastore the name of the datastore to be updated
+     * @param key       the persisted data key that has to be updated
+     * @param value     the new value
+     */
+    public static void updatePersistence(final String datastore, final String key, final String value) {
+        final Application application = TestHelper.defaultApplication;
+
+        if (application == null) {
+            fail("Unable to updatePersistence by TestPersistenceHelper. Application is null, fast failing the test case.");
+        }
+
+        final Context context = application.getApplicationContext();
+
+        if (context == null) {
+            fail("Unable to updatePersistence by TestPersistenceHelper. Context is null, fast failing the test case.");
+        }
+
+        SharedPreferences sharedPreferences = context.getSharedPreferences(datastore, Context.MODE_PRIVATE);
+
+        if (sharedPreferences == null) {
+            fail("Unable to updatePersistence by TestPersistenceHelper. sharedPreferences is null, fast failing the test case.");
+        }
+
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        editor.putString(key, value);
+        editor.apply();
+    }
+
+    static Map<String, Object> createIdentityMap(final String namespace, final String id) {
+        Map<String, Object> namespaceObj = new HashMap<>();
+        namespaceObj.put("authenticationState", "ambiguous");
+        namespaceObj.put("id", id);
+        namespaceObj.put("primary", false);
+
+        List<Map<String, Object>> namespaceIds = new ArrayList<>();
+        namespaceIds.add(namespaceObj);
+
+        Map<String, List<Map<String, Object>>> identityMap = new HashMap<>();
+        identityMap.put(namespace, namespaceIds);
+
+        Map<String, Object> xdmMap = new HashMap<>();
+        xdmMap.put("identityMap", identityMap);
+
+        return xdmMap;
+    }
+
+    static InputStream convertResourceFileToInputStream(final String name, final String fileExtension) {
+        return MessagingFunctionalTestUtils.class.getClassLoader().getResourceAsStream(name + fileExtension);
+    }
+
+    static boolean readInputStreamIntoFile(final File cachedFile, final InputStream input, final boolean append) {
+        boolean result;
```

Comment:
Let's initialize result with `false` explicitly to make it clear what is default return type.